### PR TITLE
Change macro apply for better readability

### PR DESCRIPTION
**Pull request by buttercrab**
**before**

```rust
apply!(some_layer 1, some_layer 2, ..., some_layer n, some_handler);
```

**after**

```rust
apply!(some_layer 1, some_layer 2, ..., some_layer n to some_handler);
```

This change would differentiate layer and handler with more readability.

### Changes

- [x] change `apply` from `macro_rules` to `proc_macro`
- [x] move `cubby-connect-server` to `cubby-connect-server-core`
- [x] re-export `apply` in `cubby-connect-server-core`
- [x] move tests related to `apply`
- [x] add more test about `apply`
- [x] write documentations about `apply`
- [x] change ci

### CODE REVIEW

File: server/server-core/src/fn_layer.rs [hunk 1]
```diff
@@ -31,12 +31,12 @@
...
 //! // or
 //!
-//! let e = connect(fn_layer(echo), fn_handler(print)).await?;
+//! let e = connect(echo, print).await?;
 //! e.call("Hello, World!").await?;
 //!
 //! // or
-//!
-//! let e = apply!(echo, print);
+//!
```
Comment (utilForever) [inline server/server-core/src/fn_layer.rs:38]: Remove whitespace.

**Review by utilForever — APPROVED** [review]
LGTM. 👼

**Review by minchurl — APPROVED** [review]
LGTM TOO :)

**Pull request merged**